### PR TITLE
move python packages owned by makefu into separate folders in python-modules

### PR DESCRIPTION
**Pull request by makefu**
###### Motivation for this change

Move legacy python packages maintained by myself from `python-packages.nix` to `python-modules/`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

/cc @FRidh

### CODE REVIEW

File: pkgs/development/python-modules/d2to1/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ buildPythonPackage
+, lib
+, fetchPypi
+, nose
+}:
+buildPythonPackage rec {
+  pname = "d2to1";
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/d2to1/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/d2to1/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  version = "0.2.11";
+
+  buildInputs = [ nose ];
```
Comment (FRidh) [inline pkgs/development/python-modules/d2to1/default.nix]: checkInputs

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
Thanks!

**Comment by makefu** [comment]
@GrahamcOfBorg build xstatic-pygments xstatic-jquery-ui xstatic-jquery-file-upload xstatic-jquery xstatic-bootstrap xstatic-bootbox xstatic d2to1

File: pkgs/development/python-modules/d2to1/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  checkInputs = [ nose ];
+
+  src = fetchPypi {
+    inherit version pname;
+    sha256 = "1a5z367b7dpd6dgi0w8pymb68aj2pblk8w04l2c8hibhj8dpl2b4";
+  };
+
+  meta = with lib;{
+    description = "Support for distutils2-like setup.cfg files as package metadata";
+    homepage = https://pypi.python.org/pypi/d2to1;
```
Comment (dotlambda) [inline pkgs/development/python-modules/d2to1/default.nix]: https://github.com/embray/d2to1

File: pkgs/development/python-modules/xstatic-bootbox/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  version = "4.3.0.1";
+
+  src = fetchPypi {
+    inherit version pname;
+    sha256 = "0wks1lsqngn3gvlhzrvaan1zj8w4wr58xi0pfqhrzckbghvvr0gj";
+  };
+
+  meta = with lib;{
+    homepage = http://bootboxjs.com;
+    description = "bootboxjs packaged static files for python";
```
Comment (dotlambda) [inline pkgs/development/python-modules/xstatic-bootbox/default.nix]: Capitalize, as well for the other descriptions

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/d2to1/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ buildPythonPackage
+, lib
+, fetchPypi
+, nose
+, numpy
```
Comment (dotlambda) [inline pkgs/development/python-modules/d2to1/default.nix]: not used

**Review by dotlambda — CHANGES_REQUESTED** [review]
I guess most of these packages don't implement tests. Please add `doCheck = false` and a comment in these cases.

**Comment by makefu** [comment]
@GrahamcOfBorg build python2Packages.d2to1 python3Packages.d2to1

File: pkgs/development/python-modules/xstatic-bootbox/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+buildPythonPackage rec {
+  pname = "XStatic-Bootbox";
+  version = "4.3.0.1";
+
+  src = fetchPypi {
+    inherit version pname;
+    sha256 = "0wks1lsqngn3gvlhzrvaan1zj8w4wr58xi0pfqhrzckbghvvr0gj";
+  };
+
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/xstatic-bootbox/default.nix]: `# no tests implemented`

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.d2to1 python3.pkgs.d2to1 python2.pkgs.xstatic python3.pkgs.xstatic python2.pkgs.xstatic-bootbox python3.pkgs.xstatic-bootbox python2.pkgs.xstatic-bootstrap python3.pkgs.xstatic-bootstrap python2.pkgs.xstatic-jquery-file-upload python3.pkgs.xstatic-jquery-file-upload python2.pkgs.xstatic-jquery-ui python3.pkgs.xstatic-jquery-ui python2.pkgs.xstatic-pygments python3.pkgs.xstatic-pygments

File: pkgs/development/python-modules/d2to1/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{ buildPythonPackage
+, lib
+, fetchPypi
+, nose
+}:
+buildPythonPackage rec {
+  pname = "d2to1";
+  version = "0.2.11";
+
+  checkInputs = [ nose ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/d2to1/default.nix]: > Ran 0 tests in 0.002s

You may need to explicitly set `checkPhase = "nosetests"` or something like that.

Comment (makefu) [reply pkgs/development/python-modules/d2to1/default.nix]: it seems that only the github sources contain tests (not the pypi release). however tests fail on 0.2.11, i will also bump the version in a separate commit.

**Comment by makefu** [comment]
@GrahamcOfBorg build python2.pkgs.colour python3.pkgs.colour

**Comment by makefu** [comment]
@GrahamcOfBorg build python2.pkgs.d2to1 python3.pkgs.d2to1

**Review by dotlambda — APPROVED** [review]
Thanks a lot!

**Pull request merged**